### ISSUE TITLE
AppInstance.FindOrRegisterForKey("main").IsCurrent is always true

### ISSUE THREAD

**Issue by JohnCido** [microsoft/WindowsAppSDK#1832]
### Describe the bug

I was trying to use the code in [this repo](https://github.com/andrewleader/RedirectActivationWinUI3Sample) to add single instance functionality to my app. But as described in the title. The `AppInstance.FindOrRegisterForKey("main").IsCurrent` is always true. So I can't use this to terminate the existing app instance at all.

### Steps to reproduce the bug

1. Create a new Solution using WinUI 3 single package template
2. Replace part of codes in `App.xaml.cs` with the following:
```C#
protected override async void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
{
    // Get the activation args
    var appArgs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();

    // Get or register the main instance
    var mainInstance = Microsoft.Windows.AppLifecycle.AppInstance.FindOrRegisterForKey("main");

    // If the main instance isn't this current instance
    if (!mainInstance.IsCurrent)
    {
        // Redirect activation to that instance
        await mainInstance.RedirectActivationToAsync(appArgs);

        // And exit our instance and stop
        System.Diagnostics.Process.GetCurrentProcess().Kill();
        return;
    }

    // Otherwise, register for activation redirection
    Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().Activated += App_Activated;

    m_window = new MainWindow();
    m_window.Activate();
}

private Window m_window;

private void App_Activated(object sender, Microsoft.Windows.AppLifecycle.AppActivationArguments e)
{

}
```
3. Set a breakpoint at ` if (!mainInstance.IsCurrent)` line.
4. F5, then you will see `IsCurrent` with a `true` value
5. Launch the app again using the Start Menu icon
6. The `IsCurrent` value is still `true`

### Expected behavior

It should not launch a second app instance just like the solution from [this repo](https://github.com/andrewleader/RedirectActivationWinUI3Sample).

### Screenshots

![屏幕截图 2021-11-22 002311](https://user-images.githubusercontent.com/7105485/142770300-c348cc6b-e0d6-4cdb-9f75-3043eae66f16.png)

### NuGet package version

1.0.0

### Packaging type

Packaged (MSIX)

### Windows version

Windows 11 (22000)

### IDE

Visual Studio 2022

### Additional context

Tried both .NET 5 and 6, none of them are working. Should not be a Framework related issue.

**Comment by andrewleader** [comment]
Hey @JohnCido, are you compiling your app for x86? There's a known bug in 1.0.0 where `FindOrRegister` always just returns your current instance (exactly what you described) as seen in: https://github.com/microsoft/WindowsAppSDK/issues/1717

We're going to fix that in a 1.0.1 servicing release, but we don't have a plan for when that will be released yet.

First **uninstall and then reinstall your app**, then try switching it to x64. I'm going to close this bug as a dupe of that, but if switching to x64 doesn't fix your issue, you can reopen!

<img width="314" alt="image" src="https://user-images.githubusercontent.com/13246069/142923963-909df876-2afd-424b-9e3d-983ab00ca18c.png">

**Comment by JohnCido** [comment]
> Hey @JohnCido, are you compiling your app for x86? There's a known bug in 1.0.0 where `FindOrRegister` always just returns your current instance (exactly what you described) as seen in: #1717
> 
> We're going to fix that in a 1.0.1 servicing release, but we don't have a plan for when that will be released yet.
> 
> First **uninstall and then reinstall your app**, then try switching it to x64. I'm going to close this bug as a dupe of that, but if switching to x64 doesn't fix your issue, you can reopen!
> 
> <img alt="image" width="314" src="https://user-images.githubusercontent.com/13246069/142923963-909df876-2afd-424b-9e3d-983ab00ca18c.png">

Oh my god, I finally located the cause of the issue. The actual building target for my x64 configs is actually set to x86 by mistake in the config manager. Now it finally works! I close the related discussion thread too.
Thanks for the help.
